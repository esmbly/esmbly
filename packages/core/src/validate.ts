import { RunConfig } from '@esmbly/types';
import * as errors from './errors';

export default function validate(config?: RunConfig): void {
  if (!config || !config.input || !config.transformers || !config.output) {
    throw new Error(errors.MissingConfig());
  }
  if (config.input.length < 1) {
    throw new Error(errors.NoInput());
  }
  if (config.transformers.length < 1) {
    throw new Error(errors.NoTransformers());
  }
  if (config.output.length < 1) {
    throw new Error(errors.NoOutput());
  }
}
