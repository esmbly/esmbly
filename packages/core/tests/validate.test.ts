import validate from '../src/validate';
import mockConfig from './__fixtures__/config';

describe('validate', () => {
  it('throws an error if config object is missing', () => {
    expect(() => {
      validate();
    }).toThrowErrorMatchingSnapshot();
  });

  it('throws an error if no files are provided', () => {
    expect(() => {
      validate({
        ...mockConfig,
        input: [],
      });
    }).toThrowErrorMatchingSnapshot();
  });

  it('throws an error if no transformers are provided', () => {
    expect(() => {
      validate({
        ...mockConfig,
        transformers: [],
      });
    }).toThrowErrorMatchingSnapshot();
  });

  it('throws an error if no output formats are provided', () => {
    expect(() => {
      validate({
        ...mockConfig,
        output: [],
      });
    }).toThrowErrorMatchingSnapshot();
  });

  it('does not throw an error when passed a correct config object', () => {
    expect(() => {
      validate(mockConfig);
    }).not.toThrow();
  });
});
