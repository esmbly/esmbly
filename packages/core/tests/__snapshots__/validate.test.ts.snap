// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate throws an error if config object is missing 1`] = `
"Please provide a valid configuration object.
    {
      input: File[];
      transformers: Transformer[];
      output: Output[];
    }
  "
`;

exports[`validate throws an error if no files are provided 1`] = `"Please specify at least one file to transform."`;

exports[`validate throws an error if no output formats are provided 1`] = `"Please specify at least one output format."`;

exports[`validate throws an error if no transformers are provided 1`] = `"Please specify at least one transformer."`;
